Sources/BedrockService/Converse/ConverseRequestBuilder.swift

```
@@ -61,14 +57,22 @@ public struct ConverseRequestBuilder {
 
     /// Creates a ConverseRequestBuilder object based of a ConverseRequestBuilder object
     public init(from builder: ConverseRequestBuilder) throws {
-        self = try ConverseRequestBuilder(with: builder.model)
+        let newBuilder = try ConverseRequestBuilder(with: builder.model)
             .withHistory(builder.history)
             .withTemperature(builder.temperature)
             .withTopP(builder.topP)
             .withMaxTokens(builder.maxTokens)
             .withStopSequences(builder.stopSequences)
             .withSystemPrompts(builder.systemPrompts)
             .withTools(builder.tools)
+        if builder.enableReasoning {
```

I would propose to remove the if and trasnform the `withXXX()` to silently ignore the call if the model does not support it (with a `logger.trace()` statement for visibility)

---

or to throw an error ?

---

The builder does not have access to the logger, should I change that?

Two alternatives that might be more consistent with the other functions:
1) add an optional Boolean parameter to `withReasoning` and only validate the feature if the parameter is true, that way in the initialiser we can simply pass the builder.enableReasoning attribute
2) create a separate (private) function that does this exact same thing but hides this option from the developer

---

I see the `try validateFeature(.reasoning)` throws an error, isn't it sufficient. 
If it is, you can can call `withReasoning()` and the error will be generated by `validateFeature()` and propagated to the caller.

---

I think I might have misunderstood. I thought you wanted me to remove the if statement in the initialiser, but if I do then how will I make sure the new builder will have the same value for enableReasoning as the original builder?

---

I made a change to remove the if. Is this the correct way?